pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +56,19 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		isIstioPod := false
+		if _, ok := pod.Annotations["sidecar.istio.io/status"]; ok {
+			isIstioPod = true
+		} else {
+			continue
+		}
+		if !isPodReady(pod) {
+			continue
+		}
 		for _, container := range pod.Spec.Containers {
-			containsPrefix := strings.Contains(container.Image, image.Prefix)
+			containsIstioPrefix := strings.Contains(container.Image, "istio") && strings.Contains(container.Image, "proxyv2")
```

Comment:
This var name does not reflect what it really keeps. There are two different checks evaluated:
1. `istio` keyword existence,
2. `proxyv2` keyword existence.
As a result, the env should not be called only as `containsIstioPrefix`. You could either rename it or run two different validation conditions.

Reply:
adjusted